sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -71,6 +71,16 @@ private[execution] sealed trait HashedRelation extends KnownSizeEstimation {
    */
   def keyIsUnique: Boolean
 
+  /**
+   * is input: Iterator[InternalRow] empty
```

Comment:
Can you expand this comment please: You can add a "Note that, the hashed relation can be empty despite the Iterator[InternalRow] being not empty since the hashed relation skips over null keys"

Reply:
will do.

Reply:
done